### PR TITLE
feat: add navigation page

### PR DESCRIPTION
**Pull request by Demian222**

### CODE REVIEW

File: src/main/js/generateStatic.js [hunk 1]
```diff
@@ -53,3 +58,36 @@ export const genEleventy = async (temp, output) => {
   await elev.write()
   fsExtra.removeSync(temp)
 }
+
+export const genNavigationPage = (contexts, output, navPage, input) => {
+  if (!navPage) return
+
+  const links = fs.readdirSync(input.split('/')[0], { withFileTypes: true })
+    .filter(d => d.isDirectory())
```
Comment (pismenskiy) [inline src/main/js/generateStatic.js:66]: ?

File: src/main/js/generateStatic.js [hunk 2]
```diff
@@ -40,6 +42,9 @@ export const generateStatics = async (contexts, _output, basePrefix) =>
     }
     return [..._m, output]
   }, [])
+  console.log('statics=', statics)
+  return contexts.map(context => {return {...context, prefix:basePrefix ? basePrefix + '/' + context.base : context.base}})
```
Comment (pismenskiy) [inline src/main/js/generateStatic.js:46]: context.map(context => ({...context, prefix:basePrefix ? basePrefix + '/' + context.base : context.base}))

File: src/main/js/generateStatic.js (hunk 1 above)
Comment (pismenskiy) [reply src/main/js/generateStatic.js:66]: d?

File: src/main/js/generateStatic.js [hunk 3]
```diff
@@ -53,3 +58,36 @@ export const genEleventy = async (temp, output) => {
...
+export const genNavigationPage = (contexts, output, navPage, input) => {
+  if (!navPage) return
+
+  const links = fs.readdirSync(input.split('/')[0], { withFileTypes: true })
+    .filter(dir => dir.isDirectory())
+    .reduce((r, dir) => {
+      return {...r, [dir.name]: []}
+    }, {})
+
+  const sort = (a, b) => a.date > b.date ? -1 : a.date === b.date ? 0 : 1
```
Comment (antongolub) [inline src/main/js/generateStatic.js:71]: utils sortContextsByDate

File: src/main/nav-page/index.html [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE html>
+<head>
+    <meta charset="utf-8">
+    <link rel="stylesheet" href="index.css">
+  <title>Страница навигации</title>
```
Comment (antongolub) [inline src/main/nav-page/index.html:5]: Tech Radar

File: src/main/nav-page/index.html [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+</head>
+<body>
+    <h1>Список радаров</h1>
```
Comment (antongolub) [inline src/main/nav-page/index.html:8]: Tech radars

**Review by antongolub — APPROVED** [review]
Let’s merge as is.

**Review by antongolub — APPROVED** [review]

**Pull request merged**